conan/internal/cache/db/cache_database.py

```diff
@@ -22,9 +22,7 @@ def __init__(self, filename):
             self._packages.create_table()
 
     def exists_prev(self, ref):
-        # TODO: This logic could be done directly against DB
-        matching_prevs = self.get_package_revisions_references(ref)
-        return len(matching_prevs) > 0
+        return self._packages.get_package_revisions_references_count(ref) > 0
```

Comment:
I see a mere 1% speedup in timings for tests that run this function

Reply:
As this is just an existence thing, maybe it can be further improved by not COUNTing, but just returning the first, or something like that?